### PR TITLE
Add support for rails version 7.0

### PR DESCRIPTION
**Pull request by kevinrobayna**
Rails 7.0 was released a while ago so we should ensure we support it.

### CODE REVIEW

File: .github/workflows/tests.yml
Comment (kevinrobayna) [inline .github/workflows/tests.yml:11]: 6.0 is reaching its end of life soon but again if it's not annoying anyone we should keep it. https://endoflife.date/rails

**Review by ZimbiX — CHANGES_REQUESTED** [review]
Cheers =)

This will need to be rebased once https://github.com/que-rb/que/pull/390 merges.

Again, let's update the Postgres versions tests to use the latest of everything else.

We're getting a lot of matrix permutations again (testing a version of something against every version of everything else). We don't have to address this now, but if a build starts to take too long to get through them all, we should probably move some out like with the PostgreSQL versions.

**Comment by oeoeaio** [comment]
I will hold off doing a patch release until this is rebased and merged.

**Comment by kevinrobayna** [comment]
Hi @ZimbiX and @oeoeaio i just rebased the branch let see what the specs say.

**Review by ZimbiX — APPROVED** [review]
Cheers

**Pull request merged**
